server/lib/parties/party-service.ts

```diff
@@ -287,12 +285,13 @@ export default class PartyService {
       })
   }
 
-  private subscribeToParty(clientSockets: ClientSocketsGroup, party: PartyRecord) {
+  private subscribeToParty(clientSockets: ClientSocketsGroup, party: PartyRecord, time: number) {
     clientSockets.subscribe<PartyInitEvent>(
       getPartyPath(party.id),
       () => ({
         type: 'init',
         party: toPartyJson(party),
+        time,
```

Comment:
Not sure that this is really safe, since this function will be called every time a new client connects, and this time feels like it's meant to be "when this action occurs" not "when this client was initially subscribed"? You probably need to pass a closure to retrieve the current time instead (which likely makes the injection approach more appealing here)

Note that I don't think this breaks any code *right now* but it feels semantically weird and likely to cause weird breakages in the future

Reply:
Hmm, not sure I understand how to do this. Does this mean the `init` event should not reflect the time a user accepted the party invite?

Also, how would I test this?

Reply:
Is the init event about a user accepting an invite? Because it sure seems to me (from the name and all) that it's about initializing a client to their current state. To me, the `time` field in an event should almost always be "the time this event was sent by the server, in server time", because we use it to sort events.

If you don't want that to be the case here, I mean, name it something else I guess, but it seems like a problem waiting to happen and not particularly useful on the client (as implemented, you could accomplish the same thing by assuming that anything that comes in an `init` event was prior to anything else you have received, because it will always sort that way with this time)

As for testing a closure-provided time, you would pass a closure that returns a constant time, it's not really different from how you were already testing. But like I said, injection is probably a cleaner approach, in which case you would inject a `TimeProvider` or a `Clock` or whatever that would return a constant value (again, fairly similar to what you already have)